### PR TITLE
fix(reth): use reth instead of geth

### PR DESCRIPTION
**Pull request by Deniallugo**
## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

File: core/lib/eth_client/src/clients/http/query.rs [hunk 1]
```diff
@@ -129,8 +129,17 @@ impl EthInterface for QueryClient {
             .web3
             .eth()
             .block(BlockId::Number(BlockNumber::Pending))
-            .await?
-            .expect("Pending block should always exist");
+            .await?;
+        let block = if let Some(block) = block {
+            block
+        } else {
+            // fallback for local testing
```
Comment (popzxc) [inline core/lib/eth_client/src/clients/http/query.rs:136]: Pls elaborate what's the difference between local and non-local L1 envs.

**Review by popzxc — COMMENTED** [review]
Nice!

File: docker-compose-gpu-runner-cuda-12-0.yml [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+    image: "ghcr.io/paradigmxyz/reth:v0.2.0-beta.1"
+    volumes:
+      - type: bind
+        source: ./volumes/reth/data
+        target: /rethdata
+      - type: bind
+        source: ./volumes/reth/logs
+        target: /rethlogs
+      - type: bind
+        source: ./etc/test_config/
```
Comment (popzxc) [inline docker-compose-gpu-runner-cuda-12-0.yml:14]: IIRC `test_config` was introduced as a folder for some test configurations. Keeping reth configuration here is counterintuitive.

I suggest doing the following:
1. Create folder `/etc/reth`
2. Create `/etc/reth/README.md` with a couple of sentences on what this folder represents (e.g. data to run reth locally, including folders and files mounted to the reth container as read-only volumes). Please check it with grammarly and check that it doesn't look clunky😅 
3. Create `/etc/reth/chaindata` with the contents that are intended to be mounted to reth.
4. Here and in other docker-compose files use `/etc/reth/chaindata`.

File: docker-compose-gpu-runner-cuda-12-0.yml [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
+  reth:
+    restart: always
+    image: "ghcr.io/paradigmxyz/reth:v0.2.0-beta.1"
+    volumes:
+      - type: bind
+        source: ./volumes/reth/data
+        target: /rethdata
+      - type: bind
+        source: ./volumes/reth/logs
+        target: /rethlogs
```
Comment (popzxc) [inline docker-compose-gpu-runner-cuda-12-0.yml:12]: Do we need to bind it? IIRC for non-mounted directories, an ephemeral storage would be created that is dropped once the container exits. This seems to be fine for reth logs -- I doubt we'll have to check them often.

File: core/lib/eth_client/src/clients/http/query.rs [hunk 4]
```diff
@@ -129,8 +129,18 @@ impl EthInterface for QueryClient {
...
             .eth()
             .block(BlockId::Number(BlockNumber::Pending))
-            .await?
-            .expect("Pending block should always exist");
+            .await?;
+        let block = if let Some(block) = block {
+            block
+        } else {
+            // Fallback for local reth. Because of artificial nature of producing blocks in local reth setup
+            // pending block could be not yet exists
```
Comment (popzxc) [inline core/lib/eth_client/src/clients/http/query.rs:137]: "there may be no pending block".

**Review by popzxc — COMMENTED** [review]
Mostly LGTM

**Review by perekopskiy — APPROVED** [review]

**Pull request merged**